src/frameworks/uikit/ui_device.rs

```diff
@@ -14,9 +16,49 @@ pub const UIDeviceOrientationPortrait: UIDeviceOrientation = 1;
 #[allow(dead_code)]
 pub const UIDeviceOrientationPortraitUpsideDown: UIDeviceOrientation = 2;
 pub const UIDeviceOrientationLandscapeLeft: UIDeviceOrientation = 3;
-#[allow(dead_code)]
 pub const UIDeviceOrientationLandscapeRight: UIDeviceOrientation = 4;
 #[allow(dead_code)]
 pub const UIDeviceOrientationFaceUp: UIDeviceOrientation = 5;
 #[allow(dead_code)]
 pub const UIDeviceOrientationFaceDown: UIDeviceOrientation = 6;
+
+#[derive(Default)]
+pub struct State {
+    current_device: Option<id>,
+    system_version: Option<id>,
+}
+
+pub const CLASSES: ClassExports = objc_classes! {
+
+  (env, this, _cmd);
+
+  @implementation UIDevice: NSObject
+
+  + (id)currentDevice {
+    if let Some(device) = env.framework_state.uikit.ui_device.current_device {
+      device
+    } else {
+      let new = env.objc.alloc_static_object(
+          this,
+          Box::new(TrivialHostObject),
+          &mut env.mem
+      );
+      env.framework_state.uikit.ui_device.current_device = Some(new);
+      new
+    }
+  }
+
+  - (id) systemVersion { // NSString
+    let system_version = "2.0".to_string();
+    let new = ns_string::from_rust_string(env, system_version);
```

Comment:
You probably want to use `ns_string::get_static_str` to avoid a memory leak here. I guess the framework state here was an attempt at caching but it looks broken to me.